tests/tests/alchemy_api.rs

```diff
@@ -1,8 +1,11 @@
 #![allow(clippy::used_underscore_binding)]
```

Comment:
maybe we should implement an abstraction here? Like `AlchemyProvider`, `TracerProvider`? I don't know what you think. It would allow us to test the provider like we do in the eth_provider.

Reply:
Yes agreed good idea